Cargo.toml

```diff
@@ -35,8 +35,13 @@ autocfg = "1"
 serde = { version = "1.0", optional = true, default-features = false }
 rayon = { version = "1.0", optional = true }
 
+[dependencies.hashbrown]
+version = "0.8"
+default-features = false
+features = ["inline-more", "raw"]
```

Comment:
rustc doesn't want to use `inline-more`, so it seems like we shouldn't

Reply:
I'll benchmark the difference, at least.

I'm not sure we can actually "pollute" `std`'s feature choices, because that's a separate cargo run from where `indexmap` is used in `rustc` crates. Right now we have a different version anyway, but I'm not actually sure what will happen when those align -- will cargo try to rebuild all of `std` when we enable the "raw" feature?

Reply:
Here are my results regarding "inline-more" -- it's not an obvious win after all!

<details> 
  <summary><code>cargo benchcmp default inline-more</code></summary>
<pre>
 name                                         default ns/iter  inline-more ns/iter  diff ns/iter   diff %  speedup
 entry_indexmap_150                           3,379            2,932                        -447  -13.23%   x 1.15
 few_retain_indexmap_100_000                  464,831          548,747                    83,916   18.05%   x 0.85
 grow_fnv_indexmap_100_000                    2,493,403        2,417,127                 -76,276   -3.06%   x 1.03
 half_retain_indexmap_100_000                 1,189,542        1,174,703                 -14,839   -1.25%   x 1.01
 indexmap_clone_for_sort_s                    380,182          412,785                    32,603    8.58%   x 0.92
 indexmap_clone_for_sort_u32                  20,595           23,040                      2,445   11.87%   x 0.89
 indexmap_merge_shuffle                       371,277          367,470                    -3,807   -1.03%   x 1.01
 indexmap_merge_simple                        291,215          291,980                       765    0.26%   x 1.00
 indexmap_simple_sort_s                       2,112,893        1,990,345                -122,548   -5.80%   x 1.06
 indexmap_simple_sort_u32                     665,741          642,336                   -23,405   -3.52%   x 1.04
 indexmap_sort_s                              1,619,554        1,633,687                  14,133    0.87%   x 0.99
 indexmap_sort_u32                            539,317          538,438                      -879   -0.16%   x 1.00
 insert_indexmap_100_000                      2,186,996        2,194,670                   7,674    0.35%   x 1.00
 insert_indexmap_10_000                       196,985          190,736                    -6,249   -3.17%   x 1.03
 insert_indexmap_150                          3,048            2,917                        -131   -4.30%   x 1.04
 insert_indexmap_int_bigvalue_10_000          271,094          272,833                     1,739    0.64%   x 0.99
 insert_indexmap_str_10_000                   221,377          206,219                   -15,158   -6.85%   x 1.07
 insert_indexmap_string_10_000                931,370          857,500                   -73,870   -7.93%   x 1.09
 insert_indexmap_string_10_000                937,381          845,038                   -92,343   -9.85%   x 1.11
 iter_black_box_indexmap_10_000               3,038            3,130                          92    3.03%   x 0.97
 iter_sum_indexmap_10_000                     2,040            2,092                          52    2.55%   x 0.98
 lookup_indexmap_100_000_inorder_multi        86,897           90,024                      3,127    3.60%   x 0.97
 lookup_indexmap_100_000_multi                116,602          121,846                     5,244    4.50%   x 0.96
 lookup_indexmap_100_000_single               23               25                              2    8.70%   x 0.92
 lookup_indexmap_10_000_exist                 88,555           88,473                        -82   -0.09%   x 1.00
 lookup_indexmap_10_000_exist_string          171,322          165,636                    -5,686   -3.32%   x 1.03
 lookup_indexmap_10_000_exist_string_oneshot  143,021          138,956                    -4,065   -2.84%   x 1.03
 lookup_indexmap_10_000_noexist               70,532           71,557                      1,025    1.45%   x 0.99
 many_retain_indexmap_100_000                 1,259,902        1,354,056                  94,154    7.47%   x 0.93
 new_indexmap                                 7                6                              -1  -14.29%   x 1.17
 pop_indexmap_100_000                         1,661,312        1,701,130                  39,818    2.40%   x 0.98
 shift_remove_indexmap_100_000_few            16,781,653       16,870,955                 89,302    0.53%   x 0.99
 shift_remove_indexmap_2_000_full             2,511,175        2,729,205                 218,030    8.68%   x 0.92
 swap_remove_indexmap_100_000                 4,583,354        4,640,754                  57,400    1.25%   x 0.99
 with_capacity_10e5_indexmap                  203              201                            -2   -0.99%   x 1.01
</pre>
</details>

Reply:
I just tested with `std` updated to the same hashbrown 0.8.0. It does build two copies of `hashbrown`, but this seems OK. For `std`->`hashbrown` it has "rustc-dep-of-std" enabled, with associated unstable features. For `rustc`->`indexmap`->`hashbrown`, it has "raw" and (for now) "inline-more". AFAICS this works fine.

Reply:
Great, let's drop it, Alex Crichton delivered some good rationale for avoiding this feature anyway, no reason not to follow this decision (see hashbrown PR history).

Reply:
OK, I'll remove "inline-more". For reference, I suppose you're referring to https://github.com/rust-lang/hashbrown/pull/119